pallets/pool-system/src/lib.rs

```diff
@@ -682,6 +708,16 @@ pub mod pallet {
 						},
 					)?;
 
+				let epoch_duration = now.saturating_sub(epoch_last_closed);
+				let reserve_after_fees = T::PoolFees::prepare_disbursements(
+					pool_id,
+					FeeBucket::Top,
+					nav,
+					pool.reserve.total,
+					epoch_duration.into(),
+				);
+				pool.reserve.total = reserve_after_fees;
+
```

Comment:
Actually, if `EpochHook` is implemented by `pallet-pool-fees`, would you still need a `PoolFees` trait?

Reply:
I like this proposal. Was thinking about splitting up the `PoolFees` trait to not leak mutation methods per default. 

We would need to introduce two hooks: `OnEpochClosing` and `
OnEpochExecutionPostInvestPreRedeem`.

We should still keep a trimmed down version of the current `PoolFees` trait which allows `PoolRegistry` to add fees upon pool creation.

WDYT?

Reply:
Could you have one hook with 2 methods? `on_closing()` and `on_execution()` or similar?

The trimmed down version of `PoolFees` used by `PoolRegistry` will only be used by `PoolRegistry`, isn't? pools, no longer use `PoolFees` since this change, right?

Reply:
Thinking out loud 🤔. The information you give to the pool though `PoolMutate` is only used to later call `PoolFees` (or a future hook)?

Could make sense instead of passing all this info pool-registry -> pool-system -> pool-fees, call directly to `pool-fees` from `pool-registry`?

Reply:
Related: https://github.com/centrifuge/centrifuge-chain/pull/1633#discussion_r1423985117

Reply:
> Could you have one hook with 2 methods? `on_closing()` and `on_execution()` or similar?

Great suggestion. My mind was too quickly thinking of utilizing the generic `StatusNotification` hook introduced during foreign investments.

> The trimmed down version of `PoolFees` used by `PoolRegistry` will only be used by `PoolRegistry`, isn't? pools, no longer use `PoolFees` since this change, right?

The `PoolRegistry` currently doesn't need to know about `PoolFees`, see here [comment below](https://github.com/centrifuge/centrifuge-chain/pull/1633#discussion_r1425623725). Sorry for splitting up this conversation. Let's continue it in this thread.

Reply:
> The PoolRegistry currently doesn't need to know about PoolFees, see here https://github.com/centrifuge/centrifuge-chain/pull/1633#discussion_r1425623725. Sorry for splitting up this conversation. Let's continue it in this thread.

Ok, makes sense 👍🏻